rllib/examples/rl_modules/pretraining_rlm.py

```diff
@@ -0,0 +1,124 @@
+    type=int,
+    default=200,
+    help="The number of iterations to train.",
+)
+parser.add_argument(
+    "--stop-timesteps-pretraining",
+    type=int,
+    default=5000000,
+    help="The number of (environment sampling) timesteps to train.",
+)
+
+
+if __name__ == "__main__":
+
+    # Parse the command line arguments.
+    args = parser.parse_args()
+    # Ensure that the user has set the number of agents.
+    if args.num_agents == 0:
+        raise ValueError(
+            "This pre-training example script requires at least 1 agent. "
+            "Try setting the command line argument `--num-agents` to the "
+            "number of agents you want to use."
+        )
+
+    # Store the user's stopping criteria for the löater training run.
+    stop_iters = args.stop_iters
+    stop_timesteps = args.stop_timesteps
+    checkpoint_at_end = args.checkpoint_at_end
+    # Override these criteria for the pre-training run.
+    setattr(args, "stop_iters", args.stop_iters_pretraining)
+    setattr(args, "stop_timesteps", args.stop_timesteps_pretraining)
+    setattr(args, "checkpoint_at_end", True)
+
+    # Define out pre-training single-agent algorithm. We will use the same module
+    # configuration for the pre-training and the training.
+    config = (
+        PPOConfig()
+        # Enable the new API stack (RLModule and Learner APIs).
+        .experimental(_enable_new_api_stack=True)
+        .environment("CartPole-v1")
+        .rl_module(
+            # Use a different number of hidden units for the pre-trained module.
+            model_config_dict={"fcnet_hiddens": [64]},
+        )
+    )
+
+    # Run the pre-training.
+    results = run_rllib_example_script_experiment(config, args)
+    # Get the checkpoint path.
+    module_chkpt_path = results.get_best_result().checkpoint.path
+
+    # Create a new MARL Module using the pre-trained module for policy 0.
+    env = gym.make("CartPole-v1")
+    module_specs = {}
+    module_class = PPOTorchRLModule
+    for i in range(args.num_agents):
+        module_specs[f"policy_{i}"] = SingleAgentRLModuleSpec(
+            module_class=PPOTorchRLModule,
+            observation_space=env.observation_space,
+            action_space=env.action_space,
+            model_config_dict={"fcnet_hiddens": [32]},
+            catalog_class=PPOCatalog,
+        )
+
+    # Swap in the pre-trained module for policy 0.
+    module_specs["policy_0"] = SingleAgentRLModuleSpec(
+        module_class=PPOTorchRLModule,
+        observation_space=env.observation_space,
+        action_space=env.action_space,
+        model_config_dict={"fcnet_hiddens": [64]},
+        catalog_class=PPOCatalog,
+        # Note, we load here the module directly from the checkpoint.
+        load_state_path=module_chkpt_path,
+    )
+    marl_module_spec = MultiAgentRLModuleSpec(module_specs=module_specs)
+
+    # Register our environment with tune if we use multiple agents.
+    if args.num_agents > 0:
```

Comment:
Is this if-block needed? We assert that this command line arg is >0 above.

Reply:
Yeah I guess we can remove this here. Good catch @sven1977 !

Reply:
Great catch! I removed this in the follow-up commit.